modules/metrics/src/main/java/com/spotify/apollo/metrics/RequestMetrics.java

```diff
@@ -19,14 +19,23 @@
  */
 package com.spotify.apollo.metrics;
 
-import com.spotify.apollo.StatusType;
+import com.spotify.apollo.Request;
+import com.spotify.apollo.Response;
+
+import okio.ByteString;
 
 /**
  * Defines an interface for how to collect statistics for an individual incoming request.
  */
 public interface RequestMetrics {
 
   /**
+   * Register the incoming request
+   * @param request the incoming request
+   */
+  void incoming(Request request);
```

Comment:
If RequestMetrics can only be used for a single request, I think it could be sent to ServiceMetrics. I don't think the responsibilities of the constructor and incoming is really clear (timer is started in the constructor for example)